### ISSUE TITLE
Fastify MVC errors FST_ERR_PROMISE_NOT_FULFILLED

### ISSUE THREAD

**Issue by homcenco** [nestjs/nest#6363]
## Bug Report
```
{
   "level":50,
   "time":time,
   "pid":pid,
   "hostname":"hostname,
   "reqId":1,
   "err":{
      "type":"FastifyError",
      "message":"Promise may not be fulfilled with 'undefined' when statusCode is not 204",
      "stack":"FastifyError: Promise may not be fulfilled with 'undefined' when statusCode is not 204\n    at 
 /node_modules/fastify/lib/wrapThenable.js:30:30\n    at processTicksAndRejections (node:internal/process/task_queues:94:5)",
      "name":"FastifyError",
      "code":"FST_ERR_PROMISE_NOT_FULFILLED",
      "statusCode":500
   },
   "msg":"Promise may not be fulfilled with 'undefined' when statusCode is not 204"
}
```

## Current behavior

Fresh install and run.

## Input Code

https://github.com/nestjs/nest/tree/master/sample/17-mvc-fastify
 
## Environment

<pre><code>
Nest version: 7.5.4

 
For Tooling issues:
- Node version: 15.8.0
- Platform:  MacOs 11.2
- Npm: 7.5.3

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by homcenco** [comment]
> Please provide a minimum reproduction repository.

https://github.com/nestjs/nest/tree/master/sample/17-mvc-fastify
Its a "Fastify MVC sample" from current (master) nestjs repo. Installed and started with logger.

**Comment by kamilmysliwiec** [comment]
I can't reproduce your issue with this sample. @jmcdo29 can you give it a shot?

**Comment by homcenco** [comment]
> I can't reproduce your issue with this sample.  

Reproduction repository. 
https://github.com/homcenco/17-mvc-fastify

**Comment by jmcdo29** [comment]
Okay, so I get the error you do when I navigate to `localhost:3000/`, but the stack trace is all inside of Fastify:
<details>
<summary>Error code in JSON format</summary>
<code>
{
  "level": 50,
  "time": 1612892489002,
  "pid": 20230,
  "hostname": "jay-pop",
  "reqId": 1,
  "err": {
    "type": "FastifyError",
    "message": "Promise may not be fulfilled with 'undefined' when statusCode is not 204",
    "stack": "FastifyError: Promise may not be fulfilled with 'undefined' when statusCode is not 204\n    at /home/jay/Documents/code/help/17-mvc-fastify/node_modules/.pnpm/fastify@3.11.0/node_modules/fastify/lib/wrapThenable.js:30:30\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)",
    "name": "FastifyError",
    "code": "FST_ERR_PROMISE_NOT_FULFILLED",
    "statusCode": 500
  },
  "msg": "Promise may not be fulfilled with 'undefined' when statusCode is not 204"
}
</code>
</details>

Furthermore, the same error is received from fastify directly with this minimal setup
```ts
async function bootstrapFastify() {
  const app = fastify.fastify({ logger: true });
  app.register(require("point-of-view"), {
    engine: {
      handlebars: require("handlebars"),
    },
    templates: join(__dirname, "..", "views"),
  });

  app.get("/", async (req, reply) => {
    (reply as any).view("index.hbs", { message: "Hello world" });
  });
  await app.listen(3000, "0.0.0.0");
}

bootstrapFastify();
```
(I put this in the `main.ts` so I could easily mirror the setup you had)

It seems that the error comes from having an `async` route handler with no defined `return` statement. By doing `return reply.view` I was able to get the error to go away.

@kamilmysliwiec could we return the method in the underlying route handlers to silence this message?

**Comment by kamilmysliwiec** [comment]
Thanks @jmcdo29!

Fixed in 7.6.12
